src/StackExchange.Redis/TaskSource.cs

```diff
@@ -9,7 +9,8 @@ internal static class TaskSource
         /// </summary>
         /// <typeparam name="T">The type for the created <see cref="TaskCompletionSource{TResult}"/>.</typeparam>
         /// <param name="asyncState">The state for the created <see cref="TaskCompletionSource{TResult}"/>.</param>
-        public static TaskCompletionSource<T> Create<T>(object asyncState)
-            => new TaskCompletionSource<T>(asyncState, TaskCreationOptions.None);
+        /// <param name="options">The options to apply to the task</param>
+        public static TaskCompletionSource<T> Create<T>(object asyncState, TaskCreationOptions options = TaskCreationOptions.None)
+            => new TaskCompletionSource<T>(asyncState, options);
```

Comment:
FWIW, we could delete this helper now that it’s a wrap with no logic over built-in.

Reply:
@NickCraver  I'd rather keep it all in one place even if it is just inlined by the JIT; I might add a JIT hint, but frankly I'd expect it to inline like a beast